### PR TITLE
[Feature] Introduce encryption algorithm

### LINKED ISSUE #391: [Feature] Introduce encryption algorithms.
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/paimon-webui/issues) and found nothing similar.

### Motivation

Passwords should not appear in plaintext within configuration files or databases.

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by hdygxsj**
### Purpose
close #391 .
Integrating cryptographic algorithm APIs for password encryption throughout the project, as well as providing implementations of national cryptographic standards (Chinese domestic algorithms).
### How to use 

Configuration.
```yml
paimon:
  web:
    cryptography:
      digester:
        algorithm: sm3
      symmetric:
        algorithm: sm4
        properties:
          secret-key: ${SECRET_KEY}
```
Use Api.
```java
    @Autowire
    private SymmetricCryptoService symmetricCryptoService;
    
    @Autowire
    private DigesterService digesterService;
```
### Tests

org.apache.paimon.web.cryptography.Sm3DigesterServiceTest.
org.apache.paimon.web.cryptography.Sm4SymmetricCryptoServiceTest.

### API and Format

```java
/** DigesterService. */
public interface DigesterService {

    /** Digest. */
    String digestHex(String data);

    /** Digest with salt. */
    String digestHex(String data, String salt);
}

```
```java
/** SymmetricCryptoService. */
public interface SymmetricCryptoService {

    /** Encrypt. */
    String encrypt(String data);

    /** Decrypt. */
    String decrypt(String data);
}

```

### CODE REVIEW

File: paimon-web-plugin/paimon-web-plugin-cryptography/paimon-web-plugin-cryptography-all/pom.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.apache.paimon</groupId>
+        <artifactId>paimon-web-plugin-cryptography</artifactId>
+        <version>0.1-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>paimon-web-plugin-cryptography-all</artifactId>
```
Comment (SteNicholas) [inline paimon-web-plugin/paimon-web-plugin-cryptography/paimon-web-plugin-cryptography-all/pom.xml:33]: Why introduce this module?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/configrue/cryptography/DigesterConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+@Configuration
+@ConfigurationProperties(prefix = "paimon.web.cryptography.digester")
+public class DigesterConfiguration {
+
+    private String algorithm;
+
+    private Properties properties;
+
+    @Bean
+    public DigesterService digesterService() {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/configrue/cryptography/DigesterConfiguration.java:48]: Why does this method define in `DigesterConfiguration`? Meanwhile, why does `NoneDigesterService` have no factory? Please correctly use factory mode.

**Review by SteNicholas — CHANGES_REQUESTED** [review]

File: paimon-web-plugin/paimon-web-plugin-cryptography/paimon-web-plugin-cryptography-all/pom.xml (hunk 1 above)
Comment (hdygxsj) [reply paimon-web-plugin/paimon-web-plugin-cryptography/paimon-web-plugin-cryptography-all/pom.xml:33]: When adding plugins subsequently, to utilize the modules required by the plugins, it is not necessary to import all the plugins; importing only 'paimon-web-plugin-cryptography-all' will suffice.

File: paimon-web-plugin/paimon-web-plugin-cryptography/paimon-web-plugin-cryptography-sm/src/main/java/org/apache/paimon/web/cryptography/Sm4Exception.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.web.cryptography;
+
+/** Sm4Exception. */
+public class Sm4Exception extends RuntimeException {
```
[the remark this answers is not in the document]
Comment (hdygxsj) [reply paimon-web-plugin/paimon-web-plugin-cryptography/paimon-web-plugin-cryptography-sm/src/main/java/org/apache/paimon/web/cryptography/Sm4Exception.java:22]: SM4 encryption and decryption may throw exceptions, whereas SM3 does not.

File: paimon-web-plugin/paimon-web-plugin-cryptography/paimon-web-plugin-cryptography-all/pom.xml (hunk 1 above)
Comment (SteNicholas) [reply paimon-web-plugin/paimon-web-plugin-cryptography/paimon-web-plugin-cryptography-all/pom.xml:33]: @hdygxsj, you could use maven profile to implement this.

File: paimon-web-plugin/pom.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.apache.paimon</groupId>
+        <artifactId>paimon-webui</artifactId>
+        <version>0.1-SNAPSHOT</version>
```
Comment (SteNicholas) [inline paimon-web-plugin/pom.xml:30]:
```suggestion
        <version>${project.version}</version>
```

File: paimon-web-plugin/paimon-web-plugin-cryptography/paimon-web-plugin-cryptography-sm/src/main/java/org/apache/paimon/web/cryptography/Sm3DigesterService.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    public Sm3DigesterService(String confoundKey) {
+        Security.addProvider(new BouncyCastleProvider());
+        if (confoundKey != null) {
+            this.confoundKey = confoundKey;
+        }
+    }
+
+    @Override
+    public String digestHex(String data) {
+        String confounded = data + confoundKey;
```
Comment (SteNicholas) [inline paimon-web-plugin/paimon-web-plugin-cryptography/paimon-web-plugin-cryptography-sm/src/main/java/org/apache/paimon/web/cryptography/Sm3DigesterService.java:44]:
```suggestion
        String confounded = confoundKey == null? data : data + confoundKey;
```

**Review by SteNicholas — APPROVED** [review]

**Pull request closed without merging**